django/contrib/postgres/operations.py

```diff
@@ -75,3 +79,63 @@ class UnaccentExtension(CreateExtension):
 
     def __init__(self):
         self.name = 'unaccent'
+
+
+class AddIndexConcurrently(AddIndex):
+    """Create an index using Postgres's CREATE INDEX CONCURRENTLY syntax."""
+
+    def describe(self):
+        return re.sub(r'^Create', 'Concurrently create', super().describe())
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        self._ensure_not_in_transaction(schema_editor)
+
+        model = to_state.apps.get_model(app_label, self.model_name)
+        sql = re.sub('^CREATE INDEX', 'CREATE INDEX CONCURRENTLY', str(self.index.create_sql(model, schema_editor)))
+        schema_editor.execute(sql)
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        self._ensure_not_in_transaction(schema_editor)
+
+        model = from_state.apps.get_model(app_label, self.model_name)
+        sql = re.sub('^DROP INDEX', 'DROP INDEX CONCURRENTLY', str(self.index.remove_sql(model, schema_editor)))
+        schema_editor.execute(sql)
+
+    def _ensure_not_in_transaction(self, schema_editor):
```

Comment:
It seems a mixin would be useful to avoid duplicating shared behavior such as this method.